src/datasets/builder.py

```diff
@@ -1270,6 +1278,83 @@ def _get_examples_iterable_for_split(self, split_generator: SplitGenerator) -> E
         raise NotImplementedError()
 
 
+def _number_of_shards(gen_kwargs: dict) -> int:
```

Comment:
IIRC we use a similar code to shard IterableDataset, so maybe we can put this code in `py_utils.py` and use it there to ensure consistency.

Reply:
This code now resides in `utils/sharding.py`. Marking as resolved.